lib/dal/src/attribute/value.rs

```diff
@@ -634,95 +634,96 @@ impl AttributeValue {
-                                        AttributeValue::get_by_id(ctx, av_id).await?;
-                                    // XXX: We need to properly handle the difference between "there is
-                                    // XXX: no value" vs "the value is null", but right now we collapse
-                                    // XXX: the two to just be "null" when passing these to a function.
-                                    values.push(
-                                        attribute_value.view(ctx).await?.unwrap_or(Value::Null),
-                                    );
-                                }
+                                    .value,
+                            ]
+                        }
+                        ValueSource::Secret(secret_id) => {
+                            vec![Secret::payload_for_prototype_execution(ctx, secret_id)
+                                .await
+                                .map_err(Box::new)?]
+                        }
+                        other_source => {
+                            let mut values = vec![];
 
+                            for av_id in other_source
+                                .attribute_values_for_component_id(
+                                    ctx,
+                                    expected_source_component_id,
+                                )
+                                .await?
+                            {
+                                let attribute_value = AttributeValue::get_by_id(ctx, av_id).await?;
+                                // XXX: We need to properly handle the difference between "there is
+                                // XXX: no value" vs "the value is null", but right now we collapse
+                                // XXX: the two to just be "null" when passing these to a function.
                                 values
+                                    .push(attribute_value.view(ctx).await?.unwrap_or(Value::Null));
                             }
-                        };
 
-                    func_binding_args
-                        .entry(func_arg_name)
-                        .and_modify(|values| values.extend(values_for_arg.clone()))
-                        .or_insert(values_for_arg);
+                            values
+                        }
+                    };
+
+                func_binding_args
+                    .entry(func_arg_name)
+                    .and_modify(|values| values.extend(values_for_arg.clone()))
+                    .or_insert(values_for_arg);
+            }
+        }
+        // if we haven't found func binding args by now, let's see if there are any inferred inputs
+        // Note: a given input only takes args through inferences (i.e. frames) if it doesn't have any
+        // explicitly configured args
+
+        if (func_binding_args.is_empty()) {
+            let inferred_inputs = Self::get_inferred_input_values(ctx, attribute_value_id).await?;
```

Comment:
we used to only get inferred input values if we didn't find any attribute prototype arguments for the given prototype. Now, we will search for inferred input values if we don't find any func binding args for this actual attribute value id